// config/editProfile.js
var sql = require("mssql");

var config = {
    server: '192.168.74.1', // You can use 'localhost\\instance' to connect to named instance 
    database: 'TRAVEY',
    user: 'test2',
    password: 'test',
    port: 1433
};



exports.update = function(phone_number,shared_location,email,user_name,callback) {

	var q = "UPDATE tbl_User SET email_id='"+email+"' , shared_location='"+shared_location+"', user_name='"+user_name+"' where phone_number = "+phone_number;

    var conn = new sql.Connection(config);
    conn.connect().then(function () {
        var req = new sql.Request(conn);
        req.query(q).then(function (recordset) {
            callback({'response' : true});
            console.log("successful");
            conn.close();
        })
		.catch(function (Error) {
            callback(Error);
            conn.close();
        });
    })
	.catch(function (Error) {
        callback(Error);
        console.log(Error);
    })

}


exports.editAllowedToPost = function(phone_number,allowed_to_post,callback) {

	var q = "UPDATE tbl_User SET allowed_to_post="+allowed_to_post+" where phone_number = "+phone_number;
    var conn = new sql.Connection(config);
    conn.connect().then(function () {
        var req = new sql.Request(conn);
        req.query(q).then(function (recordset) {
            callback({'response' : true});
            console.log("successful");
            conn.close();
        })
		.catch(function (Error) {
            callback(Error);
            conn.close();
        });
    })
	.catch(function (Error) {
        callback(Error);
        console.log(Error);
    })

}
