// config/login.js
var crypto = require('crypto');
var rand = require('csprng');
var sql = require("mssql");
 
var config = {
    server: '192.168.74.1', // You can use 'localhost\\instance' to connect to named instance 
    database: 'TRAVEY',
    user: 'test2',
    password: 'test',
    port: 1433
};

exports.login = function (phone_number, password, callback) {
    
    var conn = new sql.Connection(config);
    var q_login = "select * from tbl_User where phone_number = " + phone_number + ";";    
    
    conn.connect().then(function () {
        var req = new sql.Request(conn);
        req.query(q_login).then(function (recordset) {

            console.log(recordset);

            if (recordset.length == 0) {
                callback({ 'response': false ,'res': "User/ phone number does not exists !" });
                console.log("User/ phone number does not exists !");
                conn.close();
            }
            else { 
                //callback({ 'res': recordset });
                var temp = recordset[0].salt;
                var hash_db = recordset[0].hashed_password;
                var newpass = temp + password;
                var hashed_password = crypto.createHash('sha512').update(newpass).digest("hex");
               
                if(hash_db === hashed_password){
                    callback({'response': true ,'res':"Login Sucess", 'record': recordset[0]});
                    console.log("successful");
			    }else
				    callback({'response': false ,'res':"Invalid Password"});
                conn.close();   
            }
            
        })
		.catch(function (Error) {
            callback(Error);
            console.log(Error);
            conn.close();
        });
    })
	.catch(function (Error) {
        callback(Error);
        console.log(Error);
    })

	//models.User.find({phone_number: phone_number},function(err,users){

	//	if(users.length != 0){

	//		var temp = users[0].salt;
	//		var hash_db = users[0].hashed_password;
	//		var id = users[0].token;
	//		var email = users[0].email;
	//		var u_name = users[0].user_name;
	//		var newpass = temp + password;
	//		var shared_location = users[0].shared_location;
	//		var hashed_password = crypto.createHash('sha512').update(newpass).digest("hex");
	//		var grav_url = gravatar.url(users[0].email, {s: '200', r: 'pg', d: '404'});
	//		users[0].gcm_id = gcm_id;
	//		users[0].save();
	//		console.log(users[0].gcm_id +  gcm_id);

	//		if(hash_db == hashed_password){

	//			callback({'response':"Login Sucess",'res':true,'token':id,'grav':grav_url,'email': email,'user_name':u_name,'shared_location': shared_location});

	//		}else{

	//			callback({'response':"Invalid Password",'res':false});

	//		}
	//	}else {

	//		callback({'response':"User not exist",'res':false});

	//	}
	//});
}
