// config/reportAdd.js

var sql = require("mssql");

var config = {
    server: '192.168.74.1', // You can use 'localhost\\instance' to connect to named instance 
    database: 'TRAVEY',
    user: 'test2',
    password: 'test',
    port: 1433
};

exports.reportAdd = function(phone_number,detail,tag,start_time,end_time,lat,lng,callback) {
	
	var conn = new sql.Connection(config);
    var q_report = "insert into Report (phone_number, upvotes ,downvotes, detail,tag, lat,lng,start_time, end_time) values("+phone_number+" ,"+0+" ,"+0+" ,'"+detail+"','"+tag+"',"+lat+","+lng+",'"+start_time+"','"+end_time+"')";    
    
    conn.connect().then(function () {
        var req = new sql.Request(conn);
        req.query(q_report).then(function (recordset) {
            callback({'response' : true});
            console.log("successful");
            conn.close();
        })
		.catch(function (Error) {
            callback(Error);
            conn.close();
        });
    })
	.catch(function (Error) {
        callback(Error);
        console.log(Error);
    })
}	
	


	
	
	
	
	// //console.log(start_time);
	// //console.log(end_time);
	// var newReport = new models.Report({ 
	// 	phone_number : phone_number,
	// 	upvotes : num,
	// 	downvotes : num,
	// 	detail : detail,
	// 	tag : tag,
	// 	location : [lat, long],
	// 	start_time : start_time,
	// 	end_time : end_time,
	// 	expireAt : end_time

	// });

	// newReport.save(function (err) {

	// 	var query = models.User.find({'location': {
    //   		$near: [lat,long],
    //   		$maxDistance: distance
   	// 		}
   	// 	});

   	// 	query.exec(function (err, city) {
	// 	    if (err) {
    //     		console.log(err);
    //      		throw err;
    //   		}
    //   		else {
    //   			//console.log(end_time);
    //   			//console.log(Date(end_time));
    //   			callback({'response':"Report Sucessful Added", 'usersNearby': city});
    //      		//console.log('hi '+ city);
    //      		//res.json(city);
    //   		}
   	// 	});




		


		/*var query = models.User.find({'location': {
	  						$near: [lat,long],
	  						$maxDistance: distance
	  						}
						});

						query.exec(function (err, city) {
	  						if (err) {
	    						console.log(err);
	    						throw err;
	  						}

							  else{

							  		for (var i = city.length - 1; i >= 0; i--) {
							  			var fromu = "req.body.from";
   										var fromn = "req.body.fromn";
   						
							   			var title = tag;


   										requests.send(fromn, fromu, title,city[i].gcm_id, function (found) {
      										console.log("hii");
      										//res.json(found);
      									});
							  		}
							  		
								    console.log('Cant save: Found city:' + city);
							  }
	
	 						

						});
*/

		
		//console.log("Added");
